### ISSUE TITLE
When commit_sha is not in the default branch of origin, lib/ submodules are not properly checked out

### ISSUE THREAD

**Issue by kbembedded** [flipperdevices/flipper-application-catalog#230]
For example:

The default branch is main and is on commit `A`, the feature branch is feature and is also on commit `A`. Adding a submodule in lib/, configuing `fap_private_libs` to include this, and committing it all to a feature branch resulting in commit `B` only being in feature branch. Specifying commit_sha `B` in the flipper-application-catalog `manifest.yaml` will result in compile fail whne testing with `python3 tools/bundle.py ...` because it fails to correctly recurse/clone the submodule.

However, as soon a `B` is merged in to main, everything suddenly works with the `python3 tools/bundle.py ...` command.

If this is a limitation of how the tool checks out the repo, then this should probably be documented somewhere as a limitation.

**Comment by hedger** [comment]
Please provide an example which showcases the problem.

**Comment by kbembedded** [comment]
Sure, you can modify the pokemon manifest.yaml to have the following lines:
```
    origin: https://github.com/kbembedded/testing-flipper-application-catalog-230
    commit_sha: 45d050fb596ebbb351b07b37102e34a749f503d0
```

That commit hash is on `featurebranch` but not on `main`. That commit hash added the `lib/` folder with a git submodule. `tools/bundle.py` fails.


Now, modify the `manifest.yml` again to update only the origin:
```
    origin: https://github.com/kbembedded/testing-flipper-application-catalog-230-1
```

Same hash, but now `main` and `featurebranch` are on the same commit. `tools/bundle.py` will succeed.

**Comment by hedger** [comment]
Thanks, looking for a fix.

**Comment by hedger** [comment]
Should be fixed now - 3c71a6c6285aba388ad063626101c5b478921eed - please try with latest bundler script.
